dozer-sql/src/pipeline/builder.rs

```diff
@@ -690,3 +705,65 @@ mod tests {
         assert_eq!(output_keys, expected_keys);
     }
 }
+
+#[test]
+fn test_missing_into_in_simple_from_clause() {
+    let sql = r#"SELECT a FROM B "#;
+    let result = statement_to_pipeline(sql, &mut AppPipeline::new(), None);
+    //check if the result is an error
+    assert!(result.is_err());
+    //check if the error is of type MissingIntoClause
+    let error = result.unwrap_err();
+    assert!(matches!(error, PipelineError::MissingIntoClause));
+}
+
+#[test]
+fn test_correct_into_clause() {
+    let sql = r#"SELECT a INTO C FROM B"#;
+    let result = statement_to_pipeline(sql, &mut AppPipeline::new(), None);
+    //check if the result is ok
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_missing_into_in_nested_from_clause() {
+    let sql = r#"SELECT a FROM (SELECT a from b)"#;
+    let result = statement_to_pipeline(sql, &mut AppPipeline::new(), None);
+    //check if the result is an error
+    assert!(result.is_err());
+    //check if the error is of type MissingIntoClause
+    let error = result.unwrap_err();
+    assert!(matches!(error, PipelineError::MissingIntoClause));
+}
+
+#[test]
+fn test_correct_into_in_nested_from() {
+    let sql = r#"SELECT a INTO c FROM (SELECT a from b)"#;
+    let result = statement_to_pipeline(sql, &mut AppPipeline::new(), None);
+    //check if the result is ok
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_missing_into_in_with_clause() {
+    let sql = r#"WITH tbl as (select a from B)
+    select B
+    from tbl;"#;
+    let result = statement_to_pipeline(sql, &mut AppPipeline::new(), None);
+    //check if the result is an error
+    assert!(result.is_err());
+    //check if the error is of type MissingIntoClause
+    let error = result.unwrap_err();
+    assert!(matches!(error, PipelineError::MissingIntoClause));
```

Comment:
You can use [pattern matching](https://doc.rust-lang.org/book/ch18-00-patterns.html) here to simplify this.

```rust
assert!(matches!(result, Err(PipelineError::MissingIntoClause)))
```

The `matches!(result, Err(PipelineError::MissingIntoClause))` macro is equivalent to something like
```rust
match result {
    Err(PipelineError::MissingIntoClause) => true,
    _ => false,
}
```

Reply:
I mean we can write `assert!(matches!(result, Err(PipelineError::MissingIntoClause))`;

Reply:
Yes, I meant to say to use the macro. Just wanted to explain what the macro meant

Reply:
Ah sorry, I misread your comment